.github/workflows/docs-check.yml

```diff
@@ -27,6 +29,20 @@ jobs:
         with:
           node-version: 20
 
+      - name: Filter modified codepaths
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            cli:
+              - "cli/**"
+
+      - name: Check if CLI docs need to be generated
+        if: ${{ steps.filter.outputs.cli == 'true' }}
+        run: |
+          echo "Changes detected in cli, generating docs"
+          make docs.generate
```

Comment:
To avoid magic where the published docs do not match the contents of the `docs` directory on `main`, can we fail CI with an informative error if `make docs.generate` leads to changed files (you should be able to use Git to check if it leads to uncommitted changes)?

It's an extra manual step for the developer to run and commit `make docs.generate`, but it'll be good to have eyes on that during review anyway.